client/src/App.js

```diff
@@ -12,12 +12,24 @@ import NotFoundPage from "./components/NotFoundPage.tsx";
 function App() {
   const [user, setUser] = useState(null);
 
+  if (!user) {
+    return (
+      <div className="App">
+        <NavBar setUser={setUser} user={user} />
+        <Routes>
+          <Route path="/" element={<Login setUser={setUser} />} />
+          <Route path="*" element={<NotFoundPage />} />
+        </Routes>
+      </div>
+    );
+  }
```

Comment:
☝️ This component is being returned if you're not logged in. If you hit any route beyond the root route, it'll display the 404 not found page.